modules/actions/pool/sync-swaps.ts

```diff
@@ -0,0 +1,74 @@
+import { Chain, Prisma } from '@prisma/client';
+import { prisma } from '../../../prisma/prisma-client';
+import { V3VaultSubgraphClient } from '../../sources/subgraphs';
+import _ from 'lodash';
+import { swapTransformer } from '../../sources/transformers/swap-transformer';
+import { OrderDirection, Swap_OrderBy } from '../../sources/subgraphs/balancer-v3-vault/generated/types';
+import { swapsUsd } from '../../sources/enrichers/swaps-usd';
+
+/**
+ * Adds all swaps since daysToSync to the database. Checks for latest synced swap to avoid duplicate work.
+ *
+ * @param vaultSubgraphClient
+ * @param chain
+ * @param daysToSync
+ * @returns
+ */
+export async function syncSwaps(
+    vaultSubgraphClient: V3VaultSubgraphClient,
+    chain = 'SEPOLIA' as Chain,
+    daysToSync = 30,
+): Promise<string[]> {
+    const vaultVersion = 3;
+
+    // Get latest event from the DB
+    const latestEvent = await prisma.poolEvent.findFirst({
+        where: {
+            type: 'SWAP',
+            chain: chain,
+            vaultVersion,
+        },
+        orderBy: {
+            blockNumber: 'desc',
+        },
+    });
+
+    // Get events since the latest event or limit to number or days we want to keep them in the DB
+    const since = Math.floor(+new Date(Date.now() - daysToSync * 24 * 60 * 60 * 1000) / 1000);
+    const where =
+        latestEvent?.blockTimestamp && latestEvent?.blockTimestamp > since
+            ? { blockNumber_gte: String(latestEvent.blockNumber) }
```

Comment:
initially i did it on purpose to overlap events in case we missed something, but probably doesn't make sense, because most likely all events from the same block are available atomically.. (?)

Maybe adding id_not_in would make things more predictable, i mean we would be getting only missing swaps for sure?

Reply:
Yes, events from the same block are available atomically.